sentence.go

```diff
@@ -110,54 +170,45 @@ func Checksum(s string) string {
 	return fmt.Sprintf("%02X", checksum)
 }
 
+// MustRegisterParser register a custom parser or panic
+func MustRegisterParser(t string, parser ParserCallback) {
+	if err := RegisterParser(t, parser); err != nil {
+		panic(err)
+	}
+}
+
+// RegisterParser register a custom parser
+func RegisterParser(t string, parser ParserCallback) error {
+	if _, ok := customParsers[t]; ok {
+		return fmt.Errorf("nmea: parser for prefix '%s' already exists", t)
+	}
+
+	customParsers[t] = parser
+	return nil
+}
+
 // Parse parses the given string into the correct sentence type.
 func Parse(raw string) (Sentence, error) {
 	s, err := parseSentence(raw)
 	if err != nil {
 		return nil, err
 	}
+
+	// Custom parser allow overriding of existing parsers
+	if customParserCallback, ok := customParsers[s.Type]; ok {
+		return customParserCallback(s)
+	}
+
 	if strings.HasPrefix(s.Raw, SentenceStart) {
 		// MTK message types share the same format
 		// so we return the same struct for all types.
 		switch s.Talker {
 		case TypeMTK:
 			return newMTK(s)
 		}
-		switch s.Type {
-		case TypeRMC:
-			return newRMC(s)
-		case TypeGGA:
-			return newGGA(s)
-		case TypeGSA:
-			return newGSA(s)
-		case TypeGLL:
-			return newGLL(s)
-		case TypeVTG:
-			return newVTG(s)
-		case TypeZDA:
-			return newZDA(s)
-		case TypePGRME:
-			return newPGRME(s)
-		case TypeGSV:
-			return newGSV(s)
-		case TypeHDT:
-			return newHDT(s)
-		case TypeGNS:
-			return newGNS(s)
-		case TypeTHS:
-			return newTHS(s)
-		case TypeWPL:
-			return newWPL(s)
-		case TypeRTE:
-			return newRTE(s)
-		case TypeVHW:
-			return newVHW(s)
-		case TypeDPT:
-			return newDPT(s)
-		case TypeDBT:
-			return newDBT(s)
-		case TypeDBS:
-			return newDBS(s)
+
+		if parserCallback, ok := defaultSentenceParsers[s.Type]; ok {
```

Comment:
I think custom parsers should be forced to match the entire message prefix, talker + type.